src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -416,17 +415,32 @@ def symlink_or_copy(
 
     # Verify that the mapping isn't accidentally trying to create a file in the project source through symlinks.
     # We need to ensure we're resolving symlinks for this check to be effective.
+    # We are unlikely to hit this if calling the function through bundle map, keeping it here for other future use cases outside bundle.
     resolved_dst = dst.resolve()
     resolved_deploy_root = deploy_root.resolve()
-    if resolved_deploy_root not in resolved_dst.parents:
+    dst_is_deploy_root = resolved_deploy_root == resolved_dst
+    if (not dst_is_deploy_root) and (resolved_deploy_root not in resolved_dst.parents):
         raise NotInDeployRootError(dest_path=dst, deploy_root=deploy_root, src_path=src)
 
-    if overwrite:
+    if (not dst_is_deploy_root) and overwrite:
+        # TODO Verify: We do not want to delete if dst_is_deploy_root as it may contain artifacts from other runs of bundle map
+        # But this will treat deploy_root as a special dst, as compared to another dst directory that may also have other artifacts from a prev run
```

Comment:
FYI ^ Also inviting opinions here.

Reply:
When do we rely on overwrite here ? We clear the deploy root before each bundle, so in principle the changes should be purely additive.

Reply:
IIRC from late last night, I thought of this:
```
- src: foo/baz
   dest: some/dir
- src: foo/bar
   dest: ./
```
So `output/deploy/some/dir` has some contents.
Now `foo/bar` wants to go directly in `output/deploy`, i.e. `dst = output/deploy`.
Since `overwrite` is True by default, not only do we clear `output/deploy` at the beginning of overall bundle, but also carry out `delete(dst)`, meaning that entire `output/deploy` is deleted, including `output/deploy/some/dir`. 
My disclaimer is that I may have an incorrect understanding, I am okay to remove this extra condition if that is indeed the case.

Reply:
What I am asking is why would overwrite ever need to be true? We should always create directories as needed, and overwrite files. No?

Reply:
The issue here is that with my changes, all directories could behave that way, not just deploy root.